controllers/storagecluster/external_resources.go

```diff
@@ -710,7 +710,38 @@ func getTopologyConstrainedPoolsExternalMode(data map[string]string) (string, er
 	return string(topologyConstrainedPoolsStr), nil
 }
 
-func (r *StorageClusterReconciler) enableCsiDrivers(availableSCCs []StorageClassConfiguration) error {
+// getTopologyFailureDomainConfig retrieves the topology failure domain label and values from external resources
+func (r *StorageClusterReconciler) getTopologyFailureDomainConfig(instance *ocsv1.StorageCluster) (string, string, error) {
+	data, ok := externalOCSResources[instance.UID]
+	if !ok {
+		return "", "", fmt.Errorf("unable to retrieve external resource from externalOCSResources")
+	}
+
+	// Look for the topologyFailureDomainLabel and topologyFailureDomainValues in the external resources
+	for _, d := range data {
+		if d.Kind == "StorageClass" && d.Name == cephRbdTopologyStorageClassName {
+			label, labelExists := d.Data["topologyFailureDomainLabel"]
```

Comment:
dont we need to validate these labels exists on the node before passing them to the client operator? because client operator is not validating this today and simply applying what is passed by the ocs-operator.

Reply:
We depend on customer to pass the right args when passing this through the external script. cc @parth-gr

Reply:
@Madhu-1 we can just validate that the failure domain value, but we can't validate if the pool-to-domain value mapping I provided is correct by the user.

cc @subhamkrai, probably we can get the failure domain values and match with the user input, maybe a separate PR